src/main/fc/runtime_config.c

```diff
@@ -26,8 +26,12 @@
 #include "fc/runtime_config.h"
 #include "io/beeper.h"
 
-uint8_t armingFlags = 0;
-uint8_t stateFlags = 0;
+// Set by the rx task and read by the gyro/pid task
+volatile uint8_t armingFlags = 0;
```

Comment:
I'd prefer to hide critical section into access macros ...

Reply:
OK. I'll create a CHECK_AND_ENABLE_ARMING_FLAG macro or the like to check a bit and then set another if true.

Reply:
Is it necessary? Just modifying existing macros to return previous state may be enough ...

Reply:
OK, so here's where such an operation is needed:

```
void disarm(void)
{
    // Protect this atomic access to ensure the disarming never happens twice
    taskENTER_CRITICAL();
    if (ARMING_FLAG(ARMED)) {
        DISABLE_ARMING_FLAG(ARMED);
        taskEXIT_CRITICAL();
```

I was thinking of a

`CHECK_AND_ENABLE_ARMING_FLAG(flag_to_check, flag_to_enable)
`and
`CHECK_AND_DISABLE_ARMING_FLAG(flag_to_check, flag_to_disable)
`

But that would also really need to specify if flag_to_check should be set or clear.

More general would be test and set/clear using a mask/state to match, and mask/state to apply.

`ATOMIC_CHECK_ARMING_FLAG(checkMask, checkState, updateMask, updateState)
`
Which for the above would become:

```
    // Protect this atomic access to ensure the disarming never happens twice
    if (ATOMIC_CHECK_ARMING_FLAG(ARMED, ARMED, ARMED, 0)) {
```

Reply:
Actually I'll make this more general and pass the set of flags to be checked too.

Reply:
After having a good look at ATOMIC_BLOCK() it looks compatible with FreeRTOS, so let's stick with that. The parameter to be passed to it needs to be configMAX_SYSCALL_INTERRUPT_PRIORITY in order to match the behaviour of taskENTER_CRITICAL().

Reply:
In this case just use explicit critical section in caller, macro would be too confusing/use specific ... Just separate critical section into one place (using helper variable and another if) to avoid having unlock too far away.

Reply:
`ATOMIC_BLOCK` is fine. Not sure, but maybe using alias to explicitly state that RTOS lock is involved would make future easier.
But all other uses of ATOMIC_BLOCK need to be RTOS aware too, so just mask configMAX_SYSCALL_INTERRUPT_PRIORITY with some betaflight-like #define

Reply:
Might it make more sense to modify the ATOMIC_BLOCK macro to not take the arg and use configMAX_SYSCALL_INTERRUPT_PRIORITY in all cases?

Reply:
Actually, no that's not a good idea. ATOMIC_BLOCK(configKERNEL_INTERRUPT_PRIORITY) will only block non-ISR code, and ATOMIC_BLOCK(configMAX_SYSCALL_INTERRUPT_PRIORITY) will also block interrupts. So if we want to protect between tasks, but still allow interrupts we have that option.

Reply:
I implemented ATOMIC_BLOCK where redesigning softserial code. It never got merged, mainly because it was overcomplicated (replicating lot of RTOS functionality).
But it was desirable to block only some interrupts, to keep high-priority ISR latency low.
Maybe it's overkill now and 2 priority levels may be fine